packages/wdio-sync/src/constants.js

```diff
@@ -1,2 +1,28 @@
-export const STACKTRACE_FILTER = /((wdio-sync\/)*(build\/index.js|node_modules\/fibers)|- - - - -)/g
-export const STACKTRACE_FILTER_FN = (e) => !e.match(STACKTRACE_FILTER)
+const STACK_START = /^\s+at /
+
+const STACKTRACE_FILTER = [
+    // exclude @wdio/sync from stack traces
+    'node_modules/@wdio/sync/',
+
+    // Request emit
+    ' (events.js:',
+    ' (domain.js:',
+
+    // other
+    '(internal/process/next_tick.js',
+    'new Promise (<anonymous>)',
+    'Generator.next (<anonymous>)',
+    '__awaiter ('
```

Comment:
Could we maybe exclude the middleware errors as well?

```
'middlewares.js'
```

Reply:
@WillBrock updated.

Reply:
btw, I don't really like message: `Error: Promise was rejected with the following reason: Error: throw exception in wait until` 

Going to do PR with improved messages for waitUntil (but not as part of this PR)

Reply:
:+1: